options/options.go

```diff
@@ -586,15 +586,24 @@ func (interp *interpreter) interpretOptions(
 		ElementType: descriptorType(element),
 	}
 	var remain []*descriptorpb.UninterpretedOption
+	var optNodes []*ast.OptionNode
 	var features []*ast.OptionNode
 	for _, uo := range uninterpreted {
-		if uo.Name[0].GetIsExtension() != customOpts {
+		isCustom := uo.Name[0].GetIsExtension()
+		if isCustom != customOpts {
 			// We're not looking at these this phase.
 			remain = append(remain, uo)
 			continue
 		}
+		firstName := uo.Name[0].GetNamePart()
+		if targetType == descriptorpb.FieldOptions_TARGET_TYPE_FIELD &&
+			!isCustom && (firstName == "default" || firstName == "json_name") {
+			// Field pseudo-option that we can skip and is handled elsewhere.
+			remain = append(remain, uo)
+			continue
```

Comment:
If there is a problem processing the `default` or `json_name` pseudo-option, we just leave the entry in the uninterpreted options slice. That means we'd then report a confusing error, right after complaining about the default or JSON name value, that the `FieldOptions` message has no field named `default` or `json_name`.

This addresses that and avoid the subsequent confusing errors.

(BTW, we leave them in the uninterpreted options slice to support a "lenient" mode of interpreting -- where we just interpret everything we can and leave everything else alone. This is mainly done when trying to interpret options in an unlinked descriptor proto.)